src/App.jsx

```diff
@@ -14,33 +16,78 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function getSorted(obj, way) {
+  const currentGoods = [...obj];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  currentGoods.sort((el1, el2) => {
+    switch (way) {
+      case 'is-info':
+        return el1.localeCompare(el2);
+      case 'is-success':
+        return el1.length - el2.length;
+      default:
+        return 0;
+    }
+  });
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  return currentGoods;
+}
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [selectItem, setSelectItem] = useState('');
+  const [selReverse, setselReverse] = useState('');
+  const goods = getSorted(goodsFromServer, selectItem);
+
+  function reset() {
+    setselReverse('');
+    setSelectItem('');
+  }
+
+  if (selReverse) {
+    goods.reverse();
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${selectItem === 'is-info' ? null : 'is-light'}`}
+          onClick={() => setSelectItem('is-info')}
+        >
+          Sort alphabetically
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          className={`button is-success ${selectItem === 'is-success' ? null : 'is-light'}`}
+          onClick={() => setSelectItem('is-success')}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={`button is-warning ${selReverse === 'is-warning' ? null : 'is-light'}`}
+          onClick={() =>
+            selReverse ? setselReverse('') : setselReverse('is-warning')
+          }
+        >
+          Reverse
+        </button>
+
+        {(selectItem || selReverse) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => reset()}
```

Comment:
This is a code quality issue. You don't need to wrap `reset` call in an arrow function here. You can simply pass `reset` as the onClick handler like this: `onClick={reset}`.